src/Configuration/Config.AzureKeyVault/test/AzureKeyVaultConfigurationTest.cs

```diff
@@ -149,6 +149,48 @@ public void SupportsReload()
             Assert.Equal("Value2", provider.Get("Secret1"));
         }
 
+        [Fact]
+        public void SupportsReloadOnChange()
```

Comment:
This feature needs way more tests.  Some examples:

1. Token doesn't fire when versions didn't change
2. Reload is multi page aware
3. Reload works when new key is added
4. Reload works if key is removed

`Thread.Sleep` should be avoided because it's not guaranteed that background task would finish in time making tests flaky. Prefer using the reload token with timeout.